servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/ExecutionStrategyTestServices.java

```diff
@@ -0,0 +1,270 @@
+    static final BlockingTesterService CLASS_EXEC_ID_STRATEGY_BLOCKING_SERVICE =
+            new ClassExecIdStrategyBlockingService();
+    static final BlockingTesterService CLASS_NO_OFFLOADS_STRATEGY_BLOCKING_SERVICE =
+            new ClassNoOffloadsStrategyBlockingService();
+    static final BlockingTesterService METHOD_NO_OFFLOADS_STRATEGY_BLOCKING_SERVICE =
+            new MethodNoOffloadsStrategyBlockingService();
+
+    private ExecutionStrategyTestServices() {
+        // No instances
+    }
+
+    private static String threadName() {
+        return Thread.currentThread().getName();
+    }
+
+    static final class ThreadInfo {
+        final String handleExecutorName;
+        final String handleThreadName;
+        String requestOnSubscribeThreadName = NULL;
+        String requestOnNextThreadName = NULL;
+        String responseOnSubscribeThreadName = NULL;
+        String responseOnNextThreadName = NULL;
+
+        ThreadInfo(final GrpcServiceContext ctx) {
+            this.handleExecutorName = ctx.executionContext().executor().toString();
+            this.handleThreadName = threadName();
+        }
+
+        ThreadInfo(final String handleExecutorName, final String handleThreadName,
+                   final String requestOnSubscribeThreadName, final String requestOnNextThreadName,
+                   final String responseOnSubscribeThreadName, final String responseOnNextThreadName) {
+            this.handleExecutorName = handleExecutorName;
+            this.handleThreadName = handleThreadName;
+            this.requestOnSubscribeThreadName = requestOnSubscribeThreadName;
+            this.requestOnNextThreadName = requestOnNextThreadName;
+            this.responseOnSubscribeThreadName = responseOnSubscribeThreadName;
+            this.responseOnNextThreadName = responseOnNextThreadName;
+        }
+
+        TestResponse encode() {
+            return TestResponse.newBuilder().setMessage(handleExecutorName + '#' + handleThreadName + '#' +
+                    requestOnSubscribeThreadName + '#' + requestOnNextThreadName + '#' +
+                    responseOnSubscribeThreadName + '#' + responseOnNextThreadName).build();
+        }
+
+        static ThreadInfo parse(final TestResponse response) {
+            final String[] components = response.getMessage().split("#");
+            assert components.length == 6;
+            return new ThreadInfo(components[0], components[1], components[2], components[3], components[4],
+                    components[5]);
+        }
+    }
+
+    /// Async API:
+
+    private static class EsAsyncService implements TesterService {
+
+        @Override
+        public Single<TestResponse> test(final GrpcServiceContext ctx, final TestRequest request) {
+            final ThreadInfo threadInfo = new ThreadInfo(ctx);
+            return succeeded(threadInfo)
+                    .afterOnSubscribe(__ -> threadInfo.responseOnSubscribeThreadName = threadName())
```

Comment:
I see that `Single` and `Completable` have `before/afterOnSubscribe`, but doesn't have `whenOnSubscribe` (but `Publisher` has it). If they were missed, I can add them in a follow-up.
@NiteshKant wdyt?

Also, we don't have `whenSubscriber` for all 3 async sources and `Publisher.whenSubscription`, but `before/after*` variants are exist.

Reply:
It may have got missed, creating a `good first issue` SGTM

Reply:
https://github.com/apple/servicetalk/issues/926